### PR TITLE
eps e-payment QR Code url extraction

### PR DESCRIPTION
**Pull request by a-szotyori**
When an eps e-payment QR Code is detected, the contents of the QR Code is returned as an extraction to the client without sending it to the Gini API. The extraction name is `epsPaymentQRCodeUrl`.

### How to test
Run one of the example apps and scan an eps QR Code. To generate a QR Code go [here](https://eservice.stuzza.at/en/kundenservice/eps-demoshop/eps-demoshop-zentrale-bankauswahl.html), click on `eps-Überweisung`, enter `BKAUATWW` then click `Weiter zum Online-Banking` and finally click `Mit MobileBanking App fortfahren` to see the QR Code.

### CODE REVIEW

File: ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentImpl.java [hunk 1]
```diff
@@ -862,6 +860,41 @@ public void onClick(final View v) {
         });
     }
 
+    private void handlePaymentQRCodeData() {
+        if (mPaymentQRCodeData != null) {
+            switch (mPaymentQRCodeData.getFormat()) {
+                case EPC069_12:
```
[the remark this answers is not in the document]
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentImpl.java]: I can "stack" it or _guard_ it:
```
if (mPaymentQRCodeData == null) {
    return;
}
switch (mPaymentQRCodeData.getFormat()) {
....
```

File: ginivision/src/main/java/net/gini/android/vision/internal/qrcode/EPSPaymentParser.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+public class EPSPaymentParser implements QRCodeParser<PaymentQRCodeData> {
+
+    public static final String EXTRACTION_ENTITY_NAME = "epsPaymentQRCodeUrl";
+
+    @Override
+    public PaymentQRCodeData parse(@NonNull final String qrCodeContent)
+            throws IllegalArgumentException {
+        final Uri uri = Uri.parse(qrCodeContent);
+        if (!"epspayment".equals(uri.getScheme())) {
```
[the remark this answers is not in the document]
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/internal/qrcode/EPSPaymentParser.java]: Will do 👍

File: ginivision/src/main/java/net/gini/android/vision/internal/qrcode/PaymentQRCodeData.java [hunk 3]
```diff
@@ -203,4 +223,10 @@ public PaymentQRCodeData createFromParcel(final Parcel in) {
             return new PaymentQRCodeData[size];
         }
     };
+
+    public enum Format {
+        BEZAHL_CODE,
+        EPC069_12,
```
[the remark this answers is not in the document]
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/internal/qrcode/PaymentQRCodeData.java]: Good point, I'll add it here, too!

File: ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kikettas) [reply ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentImpl.java]: I meant the cascading between different cases 😛

[the remark this answers is not in the document]
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/camera/CameraFragmentImpl.java]: I see, but that's the standard practice for handling different cases in the same way... It's just the syntax is not as nice as in Swift:
```
switch paymentQRCodeData.format {
case .EPC069_12, .BEZAHL_CODE:
    ...
case .EPS_PAYMENT:
    ...
default:
    ...
}
```

**Pull request merged**